test/federated/test_plan.py

```diff
@@ -227,4 +224,97 @@ def my_plan(data):
 
     assert (plan_res == th.tensor([-42, 24, 46])).all()
 
+    # delete remote object before websocket connection termination
+    del x_ptr
+
     server.terminate()
+
+
+def test_replace_worker_ids(hook):
```

Comment:
There's a lot going on here. Break this into multiple tests and add some comments about what each test is covering.